Assets/Scripts/Models/Inventory.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections.Generic;
+using Enums;
+
+namespace Models
+{
+    public class Inventory
+    {
+        public event EventHandler InventoryUpdate;
+        private Dictionary<InventoryItems, int> _inventory = new();
+
+
+        public Inventory()
+        {
+            foreach (InventoryItems item in Enum.GetValues(typeof(InventoryItems)))
+            {
+                _inventory.Add(item, 0);
+            }
+        }
+        
+        public Dictionary<InventoryItems, int> GetInventory()
+        {
+            return _inventory;
+        }
+
+        public int GetInventoryAmount(InventoryItems item)
+        {
+            return _inventory[item];
+        }
+
+        public void AddToInventory(InventoryItems item, int amount)
+        {
+            _inventory[item] += amount;
+            InventoryUpdate?.Invoke(this, EventArgs.Empty);
+        }
+
+        public void RemoveFromInventory(InventoryItems item, int amount)
+        {
+            if(_inventory[item] == 0) return;
+            _inventory[item] -= amount;
+            InventoryUpdate?.Invoke(this, EventArgs.Empty);
+        }
+
+        public bool CanBuild(InventoryItems item)
+        {
+            return _inventory[item] >= 10;
+        }
+
+        public new string ToString()
```

Comment:
Why do we need a custom toString implementation? Logging?

Reply:
Currently used for testing while there is no working UI